src/components/ConditionalFilter/UI/BulkSelect.test.tsx

```diff
@@ -0,0 +1,188 @@
+import { fireEvent, render } from "@testing-library/react";
+import * as React from "react";
+
+import BulkSelect, { BulkSelectProps } from "./BulkSelect";
+
+const intersectionObserverMock = () => ({
+  observe: () => null,
+  unobserve: () => null,
+});
+
+window.IntersectionObserver = jest.fn().mockImplementation(intersectionObserverMock);
+
+describe("ConditionalFilter / UI / BulkSelect", () => {
+  const defaultProps: BulkSelectProps = {
+    selected: {
+      options: [],
+      loading: false,
+      conditionValue: {
+        type: "bulkselect",
+        label: "in",
+        value: "input-1",
+      },
+      value: [],
+    },
+    onBlur: jest.fn(),
+    onFocus: jest.fn(),
+    onOptionsChange: jest.fn(),
+    dataTestId: "right-0",
+    error: false,
+    helperText: "Helper text",
+    disabled: false,
+  };
+
+  it("renders BulkSelect", () => {
+    // Arrange & Act
+    const element = render(<BulkSelect {...defaultProps} />);
+
+    // Assert
+    expect(element).toMatchSnapshot();
```

Comment:
suggestion: Snapshot tests for DOM elements are not recommended because are based on structure, not component logic. The structure might change without changing logic of the component.